moduleframework/mtf_scheduler.py

```diff
@@ -227,8 +227,8 @@ def avocado_run(self):
             ACTION=self.args.action, AVOCADO_ARGS=avocado_args)
 
         # run avocado with right cmd arguments
-        bash = process.run("{AVOCADO} {a} {b}".format(
-            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=" ".join(self.tests)), shell=True, ignore_status=True)
+        bash = subprocess.check_call("{AVOCADO} {a} {b}".format(
+            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=" ".join(self.tests)), shell=True)
         common.print_info(bash.stdout, bash.stderr)
         common.print_debug("Command used: ", bash.command)
         return bash.exit_status
```

Comment:
I would prefer to use exit_code from subprocess. https://docs.python.org/2/library/subprocess.html
And use Popen which returns return code. https://docs.python.org/2/library/subprocess.html#subprocess.Popen.returncode